.pre-commit-config.yaml

```diff
@@ -0,0 +1,119 @@
+exclude: Pipfile\.lock|migrations|\.idea|node_modules
+
+repos:
+  - repo: local # More local hooks are defined at the bottom.
+    hooks:
+      - id: i18n
+        name: i18n
+        entry: bash -c 'if [ ! -f "$(dirname "$dir")"/src/locales/ar.json ]; then pnpm i18n; fi'
+        language: system
+        pass_filenames: false
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+      - id: check-executables-have-shebangs
+      - id: check-json
+        exclude: tsconfig.json # contains comments
+      - id: check-case-conflict
+      - id: check-toml
+      - id: check-merge-conflict
+      - id: check-xml
+      - id: check-yaml
+      - id: end-of-file-fixer
+        exclude: test/tapes/.+\.json5
+      - id: check-symlinks
+      - id: mixed-line-ending
+      - id: fix-encoding-pragma
+        args:
+          - --remove
+      - id: requirements-txt-fixer
+
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.9.1
+    hooks:
+      - id: isort
+        files: \.py$
+        exclude: ^build/.*$|^.tox/.*$|^venv/.*$
+        args:
+          - --lines-after-imports=2
+          - --multi-line=3
+          - --trailing-comma
+          - --force-grid-wrap=0
+          - --use-parentheses
+          - --ensure-newline-before-comments
+          - --line-length=88
+
+  - repo: https://github.com/PyCQA/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        args:
+          - --config=automations/.flake8
+
+  - repo: https://github.com/ambv/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+        args:
+          - --safe
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
```

Comment:
Would we have to manually update the package versions of `additional_dependencies` in the future?

Reply:
Yes, that will be a [manual job](https://github.com/WordPress/openverse-frontend/pull/1862#issuecomment-1261986162).